framework/i18n/MessageFormatter.php

```diff
@@ -99,10 +99,14 @@ public function format($pattern, $params, $language)
         $pattern = $this->replaceNamedArguments($pattern, $params, $newParams);
         $params = $newParams;
 
-        $formatter = new \MessageFormatter($language, $pattern);
-        if ($formatter === null) {
-            $this->_errorCode = intl_get_error_code();
-            $this->_errorMessage = 'Message pattern is invalid: ' . intl_get_error_message();
+        try {
+            $formatter = new \MessageFormatter($language, $pattern);
+            if ($formatter === null) {
+                throw new \Exception(intl_get_error_message(), intl_get_error_code());
+            }
+        } catch (\Exception $e) {
```

Comment:
@samdark Hm, it catches \IntlException so maybe that is indeed better to use, I will amend this.
@SilverFire I'm on it.